osu.Game/Overlays/MusicController.cs

```diff
@@ -136,11 +136,20 @@ public bool TogglePause()
         }
 
         /// <summary>
-        /// Play the previous track.
+        /// Play the previous track or restart the current track if it's current time below 5000ms
         /// </summary>
         /// <returns>Whether the operation was successful.</returns>
         public bool PrevTrack()
         {
+            var currentTrackPosition = current?.Track.CurrentTime;
+
+            if (currentTrackPosition >= 5000)
+            {
+                SeekTo(0);
+
+                return true;
+            }
```

Comment:
I think this logic should go into the `case GlobalAction.MusicPrev` as it seems out of scope for what this function should do. (play previous track)

Reply:
It's not that simple, this logic is shared with `NowPlayingOverlay`, whose previous button should probably behave the same way.